### PR TITLE
documentation for redaction configuration on listeners

### PR DESCRIPTION
**Pull request by peteski22**
Documentation for per-listener redaction configuration (https://github.com/hashicorp/vault/pull/23534)

Deployed page: https://vault-git-docs-peteski22vault-19863listener-re-abd165-hashicorp.vercel.app/vault/docs/configuration/listener

### CODE REVIEW

**Review by kubawi — APPROVED** [review]
Looks good!

File: website/content/docs/configuration/listener/index.mdx
Comment (ccapurso) [inline website/content/docs/configuration/listener/index.mdx:50]: 🤔 I wonder if it would make sense to provide the settings first so that the reader has further context prior to reading through the examples. What do you think?

**Review by ccapurso — APPROVED** [review]
I have one minor question regarding ordering of sections but this looks great!

File: website/content/docs/configuration/listener/index.mdx
Comment (schavis) [inline website/content/docs/configuration/listener/index.mdx:45]:
```suggestion
Vault replaces redacted information with an empty string (`""`). APIs that
omit response parameters with empty values also suppress redacted values in
response objects.

<Note title="Redacting values affects all API responses">
  The Vault CLI and UI consume Vault API responses. As a result, your redaction
  settings will apply to CLI and UI output in addition to direct API calls.
</Note>
```

Style correction: write in active voice, avoid back-to-back asides when possible

File: website/content/docs/configuration/listener/index.mdx
Comment (schavis) [reply website/content/docs/configuration/listener/index.mdx:50]: +1, it's generally a better idea to provide instruction before the examples. Examples without context don't necessarily help the reader. Especially if the first sentence encourages the reader to jump to those instructions anyway 😅

File: website/content/docs/configuration/listener/index.mdx
Comment (schavis) [inline website/content/docs/configuration/listener/index.mdx:55]: In general, we should introduce examples with plain language that explains the outcome. In this case, we really don't need to explicitly list the relevant parameters since (1) the name makes it pretty obvious and (2) if we switch things around so the settings are first in the doc, in theory, the reader will already have an idea of what they're looking at and which parameters matter to the example :)  

```suggestion
Configure the [`tcp`](/vault/docs/configuration/listener/tcp) listener to
redact IP address, cluster name, and version information:
```

Style correction: write in the active voice, use declarative language to introduce instruction or examples.

File: website/content/docs/configuration/listener/index.mdx
Comment (schavis) [inline website/content/docs/configuration/listener/index.mdx:80]:
```suggestion
listener "tcp" {
  address             = "127.0.0.1:8200",
  tls_cert_file = "/path/to/full-chain.pem"
  tls_key_file  = "/path/to/private-key.pem"
  redact_addresses    = "true"
  redact_cluster_name = "true"
  redact_version      = "true"
}
```

In general, we should focus on the most important information in example code. The stuff outside the `listener` stanza is irrelevant to the outcome described in the example description so it makes the example more accessible if we exclude it. Sometimes extra info is useful so folks can orient themselves, but in this case, the stanza is literally called `listener "tcp"` and it's a top-level configuration block, so the additional context isn't really needed.

File: website/content/docs/configuration/listener/index.mdx
Comment (schavis) [inline website/content/docs/configuration/listener/index.mdx:83]: The headings here could create confusion because it makes it look like you're moving on to another example when, really, you're illustrating the results of the previous example. I would recommend streamlining this a bit:

1. Limit the examples to 2 calls: one that shows the redacted values returned as empty strings and one that shows the redacted values omitted from the response. 
2. Provide two example responses and label them as "without redaction" and "with redaction" so folks can see the difference in the responses for themselves.
3. Simplify the text about the differences to something like "In the unredacted response `field_1` is populated with `value_1`. In the redacted response, `field_1` is [empty | omitted]."

File: website/content/docs/configuration/listener/index.mdx
Comment (schavis) [inline website/content/docs/configuration/listener/index.mdx:153]: Similar feedback here. Having a heading here implies that the CLI example relates to configuring the listener, but it's really an illustration of what the response looks like. At minimum I would fold this into the discussion above either as a third example (if we feel the distinction between direct API and CLI calls is critical) or as one of the two examples mentioned previously.

File: website/content/docs/configuration/listener/index.mdx
Comment (peteski22) [reply website/content/docs/configuration/listener/index.mdx:50]: 😨 I missed this comment before I just hit merge! 😢 Lemme fix that.

File: website/content/docs/configuration/listener/index.mdx
Comment (schavis) [inline website/content/docs/configuration/listener/index.mdx:175]: What does N/A vs none mean in this case? I'm assuming N/A means the API returned an empty string and "none" means the field was omitted from the response, but it would be helpful for folks if we made it clear which is which.

File: website/content/docs/configuration/listener/index.mdx
Comment (schavis) [inline website/content/docs/configuration/listener/index.mdx:183]: Including this link here doesn't make sense as we're expecting folks to have already skimmed past those sections to get here.  In general, we should avoid in-page links unless it's really needed from an accessibility or discoverability perspective.

```suggestion
```

File: website/content/docs/configuration/listener/index.mdx
Comment (schavis) [inline website/content/docs/configuration/listener/index.mdx:181]: I'm not sure this sentence adds much as folks typically assume a parameter applies unless we say otherwise. Something like "Use the following parameters to configure redaction for your listeners" would be more appropriate.

File: website/content/docs/configuration/listener/index.mdx
Comment (schavis) [inline website/content/docs/configuration/listener/index.mdx:187]:
```suggestion
- `redact_addresses` `(bool: false)` - Redacts `leader_address` and `cluster_leader_address` values in applicable API responses when `true`.
```

File: website/content/docs/configuration/listener/index.mdx
Comment (schavis) [inline website/content/docs/configuration/listener/index.mdx:188]:
```suggestion
- `redact_cluster_name` `(bool: false)` - Redacts `cluster_name` values in applicable API responses when `true`.
```

File: website/content/docs/configuration/listener/index.mdx
Comment (peteski22) [reply website/content/docs/configuration/listener/index.mdx:50]: @ccapurso https://github.com/hashicorp/vault/pull/23572 👍🏼

File: website/content/docs/configuration/listener/index.mdx
Comment (schavis) [inline website/content/docs/configuration/listener/index.mdx:189]:
```suggestion
- `redact_version` `(bool: false)` - Redacts `version` and `build_date` values in applicable API responses when `true`.
```

**Review by schavis — CHANGES_REQUESTED** [review]

**Comment by schavis** [comment]
In future, please do not merge documentation changes until a member of the EDU team has had a chance to review it.

File: website/content/docs/configuration/listener/index.mdx
Comment (peteski22) [reply website/content/docs/configuration/listener/index.mdx:181]: I know what you mean, but I was trying to add something that meant the common config options section could be extended later if we had settings that weren't for just redaction (e.g. disabling an endpoint via config here).  🤔

**Pull request merged**
